### ISSUE TITLE
[a11y]: DatePicker-flatpickr-day: Content is not within a landmark element

### ISSUE THREAD

**Issue by yujinpingping** [carbon-design-system/carbon#15289]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v10.56.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

![image](https://github.com/carbon-design-system/carbon/assets/151118226/7de3c996-f6e0-4f4a-9697-dba6ea76abad)

### WCAG 2.1 Violation

WCAG technique ARIA11

### Reproduction/example

N/A

### Steps to reproduce

Go to Messaging https://dev.console.test.cloud.ibm.com/messaging
Click the Calendar

Use IBM Equal Access Accessibility Checker to scan

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by yujinpingping** [comment]
sorry, please see this screenshot:
![image](https://github.com/carbon-design-system/carbon/assets/151118226/68c1b514-414a-434c-be8f-14e9d35e2eff)

**Comment by yujinpingping** [comment]
react version is: @carbon/react@1.42.1
